clang/lib/AST/Interp/Interp.h

```diff
@@ -2735,6 +2736,13 @@ inline bool InvalidDeclRef(InterpState &S, CodePtr OpPC,
   return CheckDeclRef(S, OpPC, DR);
 }
 
+inline bool SizelessVectorElementSize(InterpState &S, CodePtr OpPC) {
```

Comment:
If you look at what the current interpreter does: 1) This should use `CCEDiag` instead and 2) the diagnostic should only be emitted if we're in a constant context

Reply:
Thanks, agree 100%, it should keeps same behavior with https://github.com/llvm/llvm-project/blob/c8c0b18b5d58415b79ea13b62eee70c130c26a0a/clang/lib/AST/ExprConstant.cpp#L14080-L14092 .